### PR TITLE
Handle Vert.x Context in MutinyEmitter.send

### PR DESCRIPTION
**Pull request by cescoffier**
When the `MutinyEmitter.send` method is called from a Vert.x context, the acknowledgement event should be emitted on the same context.

This PR captures the context in the `send` method, if there is one.
In this case, it emits the result on the same context.

### CODE REVIEW

**Comment by cescoffier** [comment]
CC @unfromage

**Review by Emily-Jiang — COMMENTED** [review]
@cescoffier Does this mean the core SmallRye Reactive Messaging require Vert.x?

**Comment by cescoffier** [comment]
@Emily-Jiang Not as mandatory, but some features use Vert.x. If you use the regular Emitter, Vert.x is not required. If you use the MutinyEmitter, then yes. Suppose you are interested in using the MutinyEmitter without Vert.x, you can propose a PR checking if Vert.x is available. 

BTW, last time I checked, Open Liberty was using an older version of SmallRye Reactive Messaging. This PR only targets the 3.x (main) branch. 2.x is not changed.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang Not as mandatory, but some features use Vert.x. If you use the regular Emitter, Vert.x is not required. If you use the MutinyEmitter, then yes. Suppose you are interested in using the MutinyEmitter without Vert.x, you can propose a PR checking if Vert.x is available.
> 
Thank you @cescoffier ! If Vert.x is not avaiable, will MutinyEitter work at all? MutinyEmitter is in the core though. If I consume the core module, without Vert.x, the module will not function. Right?
> BTW, last time I checked, Open Liberty was using an older version of SmallRye Reactive Messaging. This PR only targets the 3.x (main) branch. 2.x is not changed.
Yes, you were right. However, Open Liberty plans to consume the latest SmallRye Reactive Messaging. Therefore, I am cautious on introducing mandatory dependencies on Vert.x in the core again.

**Comment by cescoffier** [comment]
@Emily-Jiang You can still use the module without Vert.x. As I said some features won't work, but they are not part of the spec (so you can pass the TCK without Vert.x being present). This PR does not change this, there are other dependencies. If that's a concern for your integration, PRs are welcome.

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/MutinyEmitterInjectionTest.java [hunk 1]
```diff
@@ -108,6 +110,27 @@ public void testWithProcessor() {
...
+        AtomicReference<Context> reference = new AtomicReference<>();
+        await().until(() -> unis.size() == 1);
+        unis.get(0).invoke(x -> reference.set(Vertx.currentContext())).await().indefinitely();
+
+        assertThat(reference.get()).isEqualTo(context);
+
+        context = ((VertxInternal) vertx).createEventLoopContext();
+        context.runOnContext(x -> unis.add(bean.emitter().send("b")));
+        await().until(() -> unis.size() == 2);
+        unis.get(1).invoke(x -> reference.set(Vertx.currentContext())).await().indefinitely();
```
Comment (ozangunalp) [inline smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/MutinyEmitterInjectionTest.java:131]: Isn't there a second assert missing ?

**Review by ozangunalp — CHANGES_REQUESTED** [review]
The change itself seems ok to me. Although `AbstractEmitter` is still a head-scratcher.

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/MutinyEmitterInjectionTest.java (hunk 1 above)
Comment (cescoffier) [reply smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/MutinyEmitterInjectionTest.java:130]: Yes, fixed.

**Review by ozangunalp — APPROVED** [review]

**Pull request merged**